### ISSUE TITLE
I propose a new overview section in the documentation

### ISSUE THREAD

**Issue by davidradl** [pytorch/audio#3454]
### 📚 The doc issue

I am new to PyTorch and what would have been useful to me was an overview of the project, at level higher than the code.
I think that this would be an 'and' not an 'or' with the existing docs and may help users who like this sort of conceptual view  over the project.   

I suggest we have a new Overview / concepts section in the documentation. That would include:

An overview of a typical PyTorch application, main and training data, model creation, loss and optimizer , training loop and test. I know these are also standard ML concepts, but I think it would help as they tie to specific PyTorch named concepts…
An overview of the PyTorch project in terms of python modules, models, C++ libraries, overview of the generation
An overview of the packaging and how the C++ and python libraries sit together.
API overviews.
Compiler overview contrasting the newer, older and recommended approaches.
GPU acceleration overview
Testing overview and approaches.
Any other overview perspective that might be useful
A starter Glossary for main terms to be defined and linked to as the term is used
The above should contain diagrams of the concepts and also link down to the existing doc. I hope that such an overview would make it more approachable to beginners and paint the big picture.

 I am happy to put up initial content for it to be reviewed or work with others who are interested in this. I am hoping to draw lots of pictures!
  
All the sections could be built out over a series of prs.
